packages/assets-controllers/src/MultichainAssetsController/MultichainAssetsController.test.ts

```diff
@@ -0,0 +1,1539 @@
+import { ControllerMessenger } from '@metamask/base-controller';
+import type {
+  AccountAssetListUpdatedEventPayload,
+  CaipAssetTypeOrId,
+} from '@metamask/keyring-api';
+import { EthAccountType, EthMethod, EthScope } from '@metamask/keyring-api';
+import { KeyringTypes } from '@metamask/keyring-controller';
+import type { InternalAccount } from '@metamask/keyring-internal-api';
+import type { PermissionConstraint } from '@metamask/permission-controller';
+import type { SubjectPermissions } from '@metamask/permission-controller';
+import type { Snap } from '@metamask/snaps-utils';
+import { useFakeTimers } from 'sinon';
+import { v4 as uuidv4 } from 'uuid';
+
+import type {
+  AssetMetadataResponse,
+  MultichainAssetsControllerMessenger,
+  MultichainAssetsControllerState,
+} from './MultichainAssetsController';
+import {
+  getDefaultMultichainAssetsControllerState,
+  MultichainAssetsController,
+} from './MultichainAssetsController';
+import { advanceTime } from '../../../../tests/helpers';
+import type {
+  ExtractAvailableAction,
+  ExtractAvailableEvent,
+} from '../../../base-controller/tests/helpers';
+
+const mockSolanaAccount = {
```

Comment:
Just to avoid type-casting in every tests, let's force the type here, so that we know those mock accounts are "well-formed" and well-typed.

```suggestion
const mockSolanaAccount: InternalAccount = {
```